var/spack/repos/builtin/packages/rocksdb/package.py

```diff
@@ -93,6 +94,9 @@ def install(self, spec, prefix):
         env["CFLAGS"] = " ".join(cflags)
         env["PLATFORM_FLAGS"] = " ".join(ldflags)
 
+        if "+werror" not in self.spec:
```

Comment:
if this is tied to newer versions of GCC, would it be better to check that instead of making it a variant?  Or do you want to make this optional in general?

Reply:
I think this should be general. While I hit this with new GCC, being able to turn off warning to Error probably helps on other compilers which may generate warnings the devs haven't seen before.